File: CS/AppearancePrioritySample/MyGridControl/MyGridViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using System.Diagnostics;

namespace GridControlDescendant
{
    public class MyGridViewInfo : GridViewInfo
    {
        public MyGridViewInfo(DevExpress.XtraGrid.Views.Grid.GridView gridView) : base(gridView) { }

        protected override DevExpress.Utils.AppearanceObject[] GetRowMixAppearances(GridCellInfo cellInfo, 
            GridRowCellState state, 
            GridRowInfo bi, 
            DevExpress.Utils.AppearanceObjectEx column, 
            DevExpress.Utils.AppearanceObject focusedCell, 
            DevExpress.Utils.AppearanceObjectEx conditionCell)
        {
            DevExpress.Utils.AppearanceObject[] apps = base.GetRowMixAppearances(cellInfo, state, bi, column, focusedCell, conditionCell);

            if ((state & GridRowCellState.Selected) != 0)
                apps.SetValue(PaintAppearance.SelectedRow, 0);

            return apps;
        }
    }
}

File: VB/AppearancePrioritySample/MyGridControl/MyGridViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports System.Diagnostics

Namespace GridControlDescendant
	Public Class MyGridViewInfo
		Inherits GridViewInfo
		Public Sub New(ByVal gridView As DevExpress.XtraGrid.Views.Grid.GridView)
			MyBase.New(gridView)
		End Sub

		Protected Overrides Function GetRowMixAppearances(ByVal cellInfo As GridCellInfo, ByVal state As GridRowCellState, ByVal bi As GridRowInfo, ByVal column As DevExpress.Utils.AppearanceObjectEx, ByVal focusedCell As DevExpress.Utils.AppearanceObject, ByVal conditionCell As DevExpress.Utils.AppearanceObjectEx) As DevExpress.Utils.AppearanceObject()
			Dim apps() As DevExpress.Utils.AppearanceObject = MyBase.GetRowMixAppearances(cellInfo, state, bi, column, focusedCell, conditionCell)

			If (state And GridRowCellState.Selected) <> 0 Then
				apps.SetValue(PaintAppearance.SelectedRow, 0)
			End If

			Return apps
		End Function
	End Class
End Namespace
